Source/Details/_ASDisplayViewAccessiblity.mm

```diff
@@ -252,43 +252,26 @@ @implementation _ASDisplayView (UIAccessibilityContainer)
 
 #pragma mark - UIAccessibility
 
-- (void)setAccessibleElements:(NSArray *)accessibleElements
+- (void)setAccessibilityElements:(NSArray *)accessibilityElements
 {
-  _accessibleElements = nil;
+  _accessibilityElements = nil;
 }
 
-- (NSArray *)accessibleElements
+- (NSArray *)accessibilityElements
 {
   ASDisplayNode *viewNode = self.asyncdisplaykit_node;
   if (viewNode == nil) {
     return @[];
   }
   
-  if (_accessibleElements != nil) {
-    return _accessibleElements;
+  if (_accessibilityElements == nil) {
+    NSMutableArray *accessibilityElements = [[NSMutableArray alloc] init];
+    CollectAccessibilityElementsForView(self, accessibilityElements);
+    SortAccessibilityElements(accessibilityElements);
+    _accessibilityElements = accessibilityElements;
   }
   
-  NSMutableArray *accessibleElements = [[NSMutableArray alloc] init];
-  CollectAccessibilityElementsForView(self, accessibleElements);
-  SortAccessibilityElements(accessibleElements);
-  _accessibleElements = accessibleElements;
-  
-  return _accessibleElements;
-}
-
-- (NSInteger)accessibilityElementCount
```

Comment:
Hey! Are you sure we can remove this accessors? I created a sample project and tried to verify that, but all of these accessors do not call through to the `accessibilityElements` getter. I also looked into Hopper and the default implementation just returns 0 for e.g. `accessibilityElementCount`.

<img width="386" alt="screen shot 2018-08-09 at 7 28 56 am" src="https://user-images.githubusercontent.com/25288/43905463-fe243a9a-9ba5-11e8-9b7a-9176367929db.png">

Reply:
Hi @maicki , thanks for checking! I wrote a small unit test to test the accessors, and they appear to work fine. I've added those tests here. Could you cross-reference with your sample project and see if there's anything I'm missing in the tests?

Reply:
@jiawernlim I just ran your test and it seems to be failing:
<img width="1099" alt="screen shot 2018-08-09 at 7 08 24 pm" src="https://user-images.githubusercontent.com/25288/43935195-b7eb3792-9c07-11e8-8558-8cbd13c638d7.png">

Here is also the example project that shows the behavior I said before:
[iOSPlayground.zip](https://github.com/TextureGroup/Texture/files/2276354/iOSPlayground.zip)

Also the part of the Apple documentation that describes the `UIAccessibilityContainer` API:
https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/iPhoneAccessibility/Making_Application_Accessible/Making_Application_Accessible.html#//apple_ref/doc/uid/TP40008785-CH102-SW10

Reply:
@maicki Lim and I have found some pretty crazy behavior here.

If accessibility is enabled, say by turning on the accessibility inspector, the system links the private `UIAccessibility` framework, which overrides these methods to be based on `accessibilityElements`:

<img width="396" alt="screen shot 2018-08-10 at 10 27 22 am" src="https://user-images.githubusercontent.com/2466893/43972050-004b03fc-9c88-11e8-8d57-374946f79f63.png">

So basically, you get 0 if no accessibility and otherwise you get the array.

And, these implementations are faster than the naïve implementations of `accessibilityElementAtIndex:` and friends, which we can't get at if we override them. So there's that.

Also, the documentation on `-accessibilityElements` says explicitly `This can be used as an alternative to implementing the dynamic methods.` implying that it's OK to do either.

So if we remove them, we get better performance but people calling them directly when accessibility is disabled will get 0.

One crazy option is to use `#if DEBUG` and if someone calls them when accessibility is disabled (could detect by using the public functions or by searching the runtime for the private `NSObjectAccessibility` accessibility class) then we make an assertion. That gets us the production performance and some safety. Thoughts?

Reply:
Crazy town! Based on that let's move forward with the diff. Less code in this area is better in the first place.

We should put some safety in though. We could use `UIAccessibilityIsVoiceOverRunning` for detecting (not a fan of the private class) and throw an assertion in this place.

Reply:
Thanks for the suggestion @maicki ! However, it appears `UIAccessibilityIsVoiceOverRunning` returns `NO` even though the accessibility inspector is on. When running your little example project with the accessibility inspector on, `displayView.accessibilityElementCount` still returns 2 while `UIAccessibilityIsVoiceOverRunning` returns NO. 

Do you have any other suggestions on how to better detect if it's running with accessibility enabled?

Reply:
Hello @maicki, just checking in to see if you have any suggestions on better ways to determine if accessibility is on. We could definitely just proceed with using `UIAccessibilityIsVoiceOverRunning`, but as mentioned earlier it will still fail when developers have the accessibility inspector on but voiceover off. We could also go with the option that Adlai had suggested, which was to search for the private class, but I agree that's not an elegant solution either. 

What are your thoughts?

Reply:
Hi @appleguy, the problem is that those unit tests will fail if the accessibility inspector is *not* on, and as such it'd fail on most systems by default. There does not seem to be an elegant way to test if the inspector is on, nor is there any way I have been able to find to programmatically turn on the inspector. 

That unit test could be removed, though then we would have no guarantees that they'd continue to work in the future. If we take this route, then perhaps we could add warnings in the header file "_ASDisplayViewAccessibility.h" to recommend use of `accessibilityElements` exclusively, and also mention that no guarantees are offered on the older accessors e.g. `accessibilityElementCount`. 

That being said, the [UIKit docs](https://developer.apple.com/documentation/objectivec/nsobject/1615147-accessibilityelements) quite specifically state that "Containers can implement this (`accessibilityElements`) property instead of the dynamic methods to support the retrieval of the contained elements." So realistically, the correctness of the older accessors should not be an issue as long as accessibility is enabled. Hence, I personally don't think this should be an issue even without having the unit tests, especially if we added some kind of warning in the header file.

Reply:
@jiawernlim Sorry didn't see your messages a couple of days ago, but I would say let's target get this PR landed :)

I would suggest let's comment out the unit tests and add a comment why they are commented out and add some kind of warning in the header file (as you mentioned the docs for `accessibilityElements` we should be relatively save there) and we should be good to go.